model_analyzer/analyzer.py

```diff
@@ -344,3 +351,12 @@ def _get_top_n_model_config_names(self, n=-1):
             x.run_config().model_variants_name()
             for x in self._result_manager.top_n_results(n=n)
         ]
+
+    def _do_checkpoint_gpus_match(self, gpus) -> bool:
+        server_only_data = self._result_manager.get_server_only_data()
+
+        for gpu in gpus:
```

Comment:
This makes sure that all requested GPUs are in the checkpoint, but does not confirm that they are the same.

The user could have run in docker mode with 2 GPUs visible, and later restarted with only 1 GPU visible. In that case this code will say it is a match, but it is not.

Reply:
Good point. I've updated the code to check that they exactly match.